analyses/fusion-frequencies/01-fusion-frequencies.R

```diff
@@ -0,0 +1,296 @@
+  distinct()
+# assert all ensg_ids and gene_symbols are not NA
+stopifnot(identical(sum(is.na(ensg_hugo_rmtl_df$ensg_id)), as.integer(0)))
+stopifnot(identical(sum(is.na(ensg_hugo_rmtl_df$gene_symbol)), as.integer(0)))
+# assert all ensg_id are unique
+# result SNV table is left joined by ensg_id
+stopifnot(identical(length(unique(ensg_hugo_rmtl_df$ensg_id)),
+                    nrow(ensg_hugo_rmtl_df)))
+
+
+
+# Subset independent samples in histology table --------------------------------
+message('Prepare data...')
+# td = tumor descriptor
+td_htl_dfs <- list(
+  primary_htl_df = htl_df %>%
+    filter(Kids_First_Biospecimen_ID %in%
+             primary_indp_sdf$Kids_First_Biospecimen_ID),
+  
+  relapse_htl_df = htl_df %>%
+    filter(Kids_First_Biospecimen_ID %in%
+             relapse_indp_sdf$Kids_First_Biospecimen_ID),
+  
+  overall_htl_df = htl_df %>%
+    filter(sample_type == 'Tumor')
+)
+
+# keep only samples with non-NA cancer_group and cohort
+# add cancer_group_cohort column
+td_htl_dfs <- lapply(td_htl_dfs, function(x) {
+  # assert all Kids_First_Biospecimen_IDs are unique
+  stopifnot(identical(nrow(x), length(unique(x$Kids_First_Biospecimen_ID))))
+  
+  fx <- filter(x, !is.na(cancer_group), !is.na(cohort))
+  fx <- mutate(
+    fx, cancer_group_cohort = paste(cancer_group, cohort, sep = '___'))
+  return(fx)
+})
+
+
+
+# Subset tumor samples to filter alteration df ---------------------------
+tumor_kfbids <- htl_df %>%
+  filter(sample_type == 'Tumor') %>%
+  pull(Kids_First_Biospecimen_ID)
+
+fusion_df <- fusion_df  %>%
+  # We want to keep track of the gene symbols for each sample-fusion pair
+  dplyr::select(Sample,
+                Kids_First_Participant_ID, 
+                FusionName, 
+                Gene1A, 
+                Gene1B, 
+                Gene2A, 
+                Gene2B, 
+                Fusion_Type,
+                DomainRetainedGene1A,
+                DomainRetainedGene1B,
+                reciprocal_exists,
+                ends_with("anno")) %>%
+  mutate(Alt_ID=paste(FusionName, Fusion_Type),sep="_") %>%
+  rename("Kids_First_Biospecimen_ID"="Sample",
+         "Kinase_domain_retained_Gene1A" = "DomainRetainedGene1A",
+         "Kinase_domain_retained_Gene1B" = "DomainRetainedGene1B",
+         "Reciprocal_exists" = "reciprocal_exists")%>%
+  # replace NA to "" in columns that have NA
+  replace_na(list("Kinase_domain_retained_Gene1A"="",
+                  "Kinase_domain_retained_Gene1B"="",
+                  "Gene1A_anno"="",
+                  "Gene1B_anno"="",
+                  "Gene2A_anno"="",
+                  "Gene2B_anno"="",
+                  "Fusion_anno"=""))%>%
+  # We want a single column that contains the gene symbols
+  tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                key = gene_position, value = gene_symbol) %>%
+  filter(!is.na(gene_symbol))
+
+
+rm(tumor_kfbids)
+suppressMessages(gc(reset = TRUE))
```

Comment:
```suggestion
```

`gc()` is no longer used in the latest commit of <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/45> .

I used it before to release RAM usage on my local computer.